java/EmbeddedCampaignManagerExample.java

```diff
@@ -0,0 +1,91 @@
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.security.SecureRandom;
+import java.util.Base64;
+
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+
+public class EmbeddedCampaignManagerExample {
+  private static final String HMAC_SHA256_ALGORITHM = "HmacSHA256";
+
+  public static void main(String[] args) {
+    String baseUrl = "https://portal.rmp.moloco.cloud";
+    String platformId = "RMP_PLATFORM_ID";
+    String adAccountId = "ad-account-id";
+    String adAccountTitle = "My Ad Account";
+    String path = "/embed/sponsored-ads/cm/a/" + adAccountId;
+    String name = "Example User Name";
+    String email = "test@example.com";
+    String role = "AD_ACCOUNT_OWNER";
+    String externalUserId = "user-id";
+    String secret = "super-secret";
+    String version = "1.0.0";
+    String colorMode = "light"; // "light" | "dark" | "useDeviceSetting"
+    String language = "en"; // "en" | "ko"
+
+    try {
+      String url = createSignedRmpPortalUrl(baseUrl, path, platformId, adAccountId, adAccountTitle, name, email, role,
+          externalUserId, secret, version, colorMode, language);
+      System.out.println(url);
+
+    } catch (Exception e) {
+      System.out.println(e);
+    }
+  }
+
+  public static String createSignedRmpPortalUrl(String baseUrl, String path, String platformId, String adAccountId,
+      String adAccountTitle, String name, String email, String role, String externalUserId, String secret,
+      String version, String colorMode, String language) throws Exception {
+    // unix timestamp in seconds
+    String timestamp = Long.toString(System.currentTimeMillis() / 1000L);
+
+    // create nonce
+    SecureRandom random = new SecureRandom();
+    byte bytes[] = new byte[20];
+    random.nextBytes(bytes);
+    String nonce = Base64.getEncoder().encodeToString(bytes);
+
+    // gather params in an alphabetical order
+    String params = "";
+    params += adAccountId + "\n";
+    params += adAccountTitle + "\n";
+    params += email + "\n";
+    params += externalUserId + "\n";
+    params += name + "\n";
+    params += nonce + "\n";
+    params += path + "\n";
+    params += platformId + "\n";
+    params += role + "\n";
+    params += timestamp + "\n";
+    params += version;
```

Comment:
nit: Considering performance, concatenating strings by using `+` operator (not `+=`) or `StringBuilder` is better. However I think the code readability matters than performance...

Reply:
I think `String.join("\n", Arrays.asList("".. ))` might be more readable

Reply:
Thanks for the review! Will try to address. (My Java knowledge hasn't been updated for a long time.)

Reply:
Addressed in https://github.com/moloco-rmp-samples/sso-examples/pull/1/commits/53a00b450a0f78ed0a949c09169765cde1523038